src/game/WorldHandlers/ChatHandler.cpp

```diff
@@ -93,7 +100,10 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket& recv_data)
         SendNotification(LANG_UNKNOWN_LANGUAGE);
         return;
     }
-    if (langDesc->skill_id != 0 && !_player->HasSkill(langDesc->skill_id))
+
+    //prevent cheating, by sending LANG_UNIVERSAL
+    if ((langDesc->lang_id == LANG_UNIVERSAL && !sWorld.getConfig(CONFIG_BOOL_ALLOW_TWO_SIDE_INTERACTION_CHAT) && GetSecurity() == SEC_PLAYER) ||
+         langDesc->skill_id != 0 && !_player->HasSkill(langDesc->skill_id))
```

Comment:
I don't understand him. But i explain why i write this code.

if ((langDesc->lang_id == LANG_UNIVERSAL && !sWorld.getConfig(CONFIG_BOOL_ALLOW_TWO_SIDE_INTERACTION_CHAT) && GetSecurity() == SEC_PLAYER) ||
langDesc->skill_id != 0 && !_player->HasSkill(langDesc->skill_id))

if ((CondTier1_1 && !CondTier1_2 && CondTier1_3) ||
CondTier2_1 && !CondTier2_2)

If all conditions of Tier 1 is a true, we don't need check Tier 2 and can switch execution routine. This work, because all Tier1 conditions wrapped by brackets AND separated by newline from Tier 2 conditions. Brackets serve as micro function, returning true of false, depends on Tier 1 conditions.
If one of the conditions if Tier 1 is a false, then we check Tier 2 conditions, which finally decides can we switch execution routine or not.